datasets/code_x_glue_cc_clone_detection_big_clone_bench/code_x_glue_cc_clone_detection_big_clone_bench.py

```diff
@@ -0,0 +1,90 @@
+from typing import List
+
+import datasets
+
+from .common import TrainValidTestChild
+from .generated_definitions import DEFINITIONS
+
+
+class CodeXGlueCCCloneDetectionBigCloneBench(TrainValidTestChild):
+    _DESCRIPTION = """Given two codes as the input, the task is to do binary classification (0/1), where 1 stands for semantic equivalence and 0 for others. Models are evaluated by F1 score.
+The dataset we use is BigCloneBench and filtered following the paper Detecting Code Clones with Graph Neural Network and Flow-Augmented Abstract Syntax Tree."""
+
+    _CITATION = """@inproceedings{svajlenko2014towards,
+  title={Towards a big data curated benchmark of inter-project code clones},
+  author={Svajlenko, Jeffrey and Islam, Judith F and Keivanloo, Iman and Roy, Chanchal K and Mia, Mohammad Mamun},
+  booktitle={2014 IEEE International Conference on Software Maintenance and Evolution},
+  pages={476--480},
+  year={2014},
+  organization={IEEE}
+}
+
+@inproceedings{wang2020detecting,
+  title={Detecting Code Clones with Graph Neural Network and Flow-Augmented Abstract Syntax Tree},
+  author={Wang, Wenhan and Li, Ge and Ma, Bo and Xia, Xin and Jin, Zhi},
+  booktitle={2020 IEEE 27th International Conference on Software Analysis, Evolution and Reengineering (SANER)},
+  pages={261--271},
+  year={2020},
+  organization={IEEE}
+}"""
```

Comment:
Can you have the citation and description as global variables ?

Reply:
It will be a bit tricky, as everything is generated, and the base class is handling this. Why is it better to have a global variable ?

Reply:
The template for datasets scripts we're using has these variable as global variables. It's the same for all the datasets. Moreover we're parsing the python scripts to extract the description and citation in moon landing to show them in the huggingface.co dataset pages, and the parsing currently expects them to be in global variables.

So it's possible to support them to be non-global but it may require changes on moon landing side. We probably want to keep the parsing as simple as possible, but it's not a major issue.

What you can do is have them as both global variables and class attribute:
```python
_CITATION = """\
insert citation here
"""

class MyDataset(datasets.GeneratorBasedBuilder):
    _CITATION = _CITATION
```

this way the main class has access to the citation and the citation is also defined as a global variable.

What do you think ?